chia/wallet/trading/offer.py

```diff
@@ -374,6 +377,18 @@ def from_spend_bundle(cls, bundle: SpendBundle) -> "Offer":
     def name(self) -> bytes32:
         return self.to_spend_bundle().name()
 
+    def compress(self) -> bytes:
+        as_spend_bundle = self.to_spend_bundle()
+        mods: List[bytes] = [bytes(cs.puzzle_reveal.to_program().uncurry()[0]) for cs in as_spend_bundle.coin_spends]
+        highest_version = max(lowest_compatible_version(mods), 2)  # 2 is the version where OFFER_MOD lives
```

Comment:
This is pretty clunky. Consider simplifying, taking a version number parameter and just compressing with that version number. A separate function (which we could write later since it's not much use right now) could then iterate over all known versions and choose the one that gives the best compression with lowest version number.